### PR TITLE
feat: add modal component

### PR DESCRIPTION
**Pull request by devdeadviz**
# Change Logs
## Screenshot
Screenshot (139)
## Deploy Preview Link:
https://deploy-preview-7--arcui.netlify.app/pages/components/modal/modal.html

### CODE REVIEW

File: pages/components/modal/modal.html [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+          <a
+            class="drawer-lists flex flexAlignItemsCenter"
+            href="../input/input.html"
+          >
+            <p class="drawer-list-content">Input</p>
+          </a>
+        </div>
+      </aside>
+      <main class="main-content m-5 width-90">
+        <h1 class="mb-4">Modal</h1>
```
Comment (irajbhinde) [inline pages/components/modal/modal.html:127]: After looking into deploy preview, I suppose m-5, my-4 is being used for setting margins. You can use a better naming here as I wasn't able to figure out what that class does until I looked into the deploy preview.

**Review by irajbhinde — COMMENTED** [review]
Looks good, just didn't find the naming used for margins to be reader-friendly

**Review by akash-1318 — COMMENTED** [review]
Looks good to merge.

File: pages/components/modal/modal.html (hunk 1 above)
Comment (devdeadviz) [reply pages/components/modal/modal.html:127]: I have used the generic way that other libraries are also using, you can check bootstrap and others also.

**Pull request merged**